expo-maps/android/src/main/java/expo/modules/maps/ExpoMapsModule.kt

```diff
@@ -1,22 +1,40 @@
 package expo.modules.maps
 
+import expo.modules.core.interfaces.LifecycleEventListener
+import expo.modules.core.interfaces.services.UIManager
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
 
 
 class ExpoMapsModule : Module() {
+
+  private val googleMapViews = mutableListOf<GoogleMapsView>()
+
   override fun definition() = ModuleDefinition {
     name("ExpoMaps")
 
-    function("someGreatMethodAsync") { options: Map<String, String> ->
-      println("Hello 👋")
-      null as Any?
-    }
-
     viewManager {
-      view { SolidColorView(it) }
-      prop("color") { view: SolidColorView, color: Int ->
-        view.setColor(color)
+      view {
+        val view = GoogleMapsView(it)
+        googleMapViews.add(view)
+
+        appContext.legacyModule<UIManager>()?.registerLifecycleEventListener(
+          object : LifecycleEventListener {
+
+            override fun onHostResume() {
+              googleMapViews.forEach { it.mapView.onResume() }
+            }
+
+            override fun onHostPause() {
+              googleMapViews.forEach { it.mapView.onPause() }
+            }
+
+            override fun onHostDestroy() {
+              googleMapViews.forEach { it.mapView.onDestroy() }
+            }
+          }
+        )
```

Comment:
This block leaks all the views. Currently, you can't fix that, but I will implement one more method - `onDropViewInstance` (https://github.com/facebook/react-native/blob/725bffce51282d649b7e071873eec76c379b1ede/ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewManager.java#L159). That should be enough to unsubscribe from the listener. I would also move that logic to the view itself - maybe into `onMapReady` or into the constructor.